pkg/resource/layer_version/hooks.go

```diff
@@ -0,0 +1,80 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package layer_version
+
+import (
+	"context"
+	"sort"
+
+	// "github.com/Jeffail/gabs/v2"
+	svcapitypes "github.com/aws-controllers-k8s/lambda-controller/apis/v1alpha1"
+	svcsdk "github.com/aws/aws-sdk-go/service/lambda"
+)
+
+var (
+	_ = &svcsdk.Lambda{}
+	_ = &svcapitypes.LayerVersion{}
+)
+
+// customPreDelete deletes all the previous versions of a
+//LayerVersion except the latest version
+
+// This function is used as a sdk_delete hook, to delete
+// all the previous versions of a LayerVersion when delete API call is made
+func customPreDelete(
+	r *resource,
+	rm *resourceManager,
+	ctx context.Context,
+) error {
+	var response *svcsdk.ListLayerVersionsOutput
+	var err error
+	var version_list []*svcsdk.LayerVersionsListItem
+	var totalVersion int
+	var _ *svcsdk.DeleteLayerVersionOutput
+
+	// Getting the list of all the versions of a LayerVersion
+	input := &svcsdk.ListLayerVersionsInput{
+		LayerName: r.ko.Spec.LayerName,
+	}
+	response, err = rm.sdkapi.ListLayerVersionsWithContext(ctx, input)
+	if err != nil {
+		return err
+	}
+
+	// The above API call returns output containing list of versions as
+	//LayerVersions and a pagination token as NextMarker
+
+	// Extracting the list of versions and assigning it to a new variable
+	version_list = response.LayerVersions
+	totalVersion = len(version_list) // Number of total versions
+
+	// sorting the list in ascending order
+	sort.Slice(version_list, func(i, j int) bool {
+		return *version_list[i].Version < *version_list[j].Version
+	})
+
+	for i := 0; i < (totalVersion - 1); i++ {
+		input := &svcsdk.DeleteLayerVersionInput{
+			LayerName:     r.ko.Spec.LayerName,
+			VersionNumber: version_list[i].Version,
+		}
+		// Delete API call to delete the versions one by one
+		_, err = rm.sdkapi.DeleteLayerVersionWithContext(ctx, input)
+		if err != nil {
+			return err
+		}
```

Comment:
You need to record the API call after making an API request here, similar to `rm.metrics.RecordAPICall("DELETE", "DeleteLayerVersion", err)`

Reply:
Added